pkg/qemu/qemu.go

```diff
@@ -194,8 +196,19 @@ func Cmdline(cfg Config) (string, []string, error) {
 	args = append(args, "-cdrom", filepath.Join(cfg.InstanceDir, filenames.CIDataISO))
 
 	// Network
+	if y.Network.VDE.SocketDir != "" {
+		macAddr := y.Network.VDE.MACAddress
+		if macAddr == "" {
+			sha1 := sha1.Sum([]byte(cfg.InstanceDir))
+			// According to https://gitlab.com/wireshark/wireshark/-/blob/master/manuf
+			// no well-known MAC addresses start with 0x22.
+			hw := append(net.HardwareAddr{0x22}, sha1[0:5]...)
+			macAddr = hw.String()
+		}
+		args = append(args, "-device", fmt.Sprintf("virtio-net-pci,netdev=net0,mac=%s", macAddr))
+		args = append(args, "-netdev", fmt.Sprintf("vde,id=net0,sock=%s", y.Network.VDE.SocketDir))
+	}
 	// CIDR is intentionally hardcoded to 192.168.5.0/24, as each of QEMU has its own independent slirp network.
-	// TODO: enable bridge (with sudo?)
 	args = append(args, "-net", "nic,model=virtio")
 	args = append(args, "-net", fmt.Sprintf("user,net=192.168.5.0/24,hostfwd=tcp:127.0.0.1:%d-:22", y.SSH.LocalPort))
```

Comment:
We should specify “id=net1” here?

Reply:
Good idea. Because the ordering alone doesn't map the VDE interface to `eth0`; it is still `eth1` for me...

Reply:
> Because the ordering alone doesn't map the VDE interface to `eth0`; it is still `eth1` for me...

Adding the id didn't help; SLIRP continues to be `eth0`.